auction-backend/src/main/java/com/atlantbh/internship/auction/app/controller/DidYouMeanController.java

```diff
@@ -0,0 +1,25 @@
+package com.atlantbh.internship.auction.app.controller;
+
+import com.atlantbh.internship.auction.app.dto.search.SearchSuggestion;
+import com.atlantbh.internship.auction.app.service.DidYouMeanService;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("api/v1/search")
+public class DidYouMeanController {
+    private final DidYouMeanService didYouMeanService;
+
+    public DidYouMeanController(final DidYouMeanService didYouMeanService) {
+        this.didYouMeanService = didYouMeanService;
+    }
+
+    @GetMapping
+    public ResponseEntity<SearchSuggestion> getSearchSuggestion(@RequestParam final String searchInput) {
+        return new ResponseEntity<>(didYouMeanService.didYouMean(searchInput), HttpStatus.OK);
+    }
+}
```

Comment:
The user can attempt to search for multiple words in a single use of the feature. This implies that with each search, an API call is sent to the backend. Would it make more sense to call the API to retrieve all items and then perform the algorithm on the frontend side?

Reply:
It would make more sense to call the backend API only once, but theoretically, it would take more time to perform algorithm calculation in JavaScript than in Java.